include/picongpu/main.cpp

```diff
@@ -33,6 +33,11 @@
 #include <picongpu/simulation_defines.hpp>
 
 
+// Workaroud for Visual Studio to avoid collisions with ERROR define
+#ifdef _MSC_VER
```

Comment:
add note on boost argsparser please as in the PR description

Reply:
please also add an `ifdef`:
```C++
#ifdef _MSC_VER
#   ifdef ERROR
#       undef ERROR
#   endif
#endif
````

Reply:
Why? If the macro is not defined, undef [does nothing](http://en.cppreference.com/w/cpp/preprocessor/replace)

Reply:
> If the identifier does not have associated macro, the directive is ignored. 

True, in that case the guard as now is fine. Pls just add the comment that we know it's about the enum in boost argsparse that collides

Reply:
typo: workarou**n**d

Reply:
But it's not boost, it's in picongpu/ArgParser.hpp.
typo - thanks, will fix.